src/detect-engine-tag.c

```diff
@@ -493,6 +493,26 @@ static void TagHandlePacketHost(Host *host, Packet *p)
     }
 }
 
+static Host *GetLockedSrcHost(Packet *p)
+{
+    if (p->host_src == NULL) {
+        p->host_src = HostLookupHostFromHash(&p->src);
+    } else {
+        HostLock(p->host_src);
+    }
+    return p->host_src;
+}
+
+static Host *GetLockedDstHost(Packet *p)
+{
+    if (p->host_dst == NULL) {
+        p->host_dst = HostLookupHostFromHash(&p->dst);
```

Comment:
Do we for this case and for the same case in `src` above not need to `HostDecrUsecnt` (facilitated by `HostRelease`) during the Unlock in the caller?

Reply:
No, the packet keeps the reference to the host(s) until it is freed/recycled. Then it calls `PacketReleaseRefs` to drop the references. This way other parts of the code that use the host can do so w/o having to do another (expensive) lookup.